test/Common/HttpApiHandlerTests.cs

```diff
@@ -1592,6 +1592,98 @@ public async Task SignalEntity_Is_Success(bool hasKey, bool hasOp, bool hasConte
             Assert.Equal(HttpStatusCode.Accepted, actualResponse.StatusCode);
         }
 
+        [Theory]
+        [InlineData("example.com", null, "example.com", null)]
+        [InlineData("example.com", "https", "example.com", "https")]
+        [InlineData("custom.domain.com", null, "custom.domain.com", null)]
+        [InlineData("custom.domain.com", "https", "custom.domain.com", "https")]

```

Comment:
Non-blocking, but since `forwardedHost` and `expectedHost` are always the same (same with `forwardedProto` and `expectedProto`) you don't actually need to define both of them. Just use the same value for both the forwarded and expected values.